crates/ethereum/evm/src/execute.rs

```diff
@@ -451,6 +451,10 @@ where
         self.batch_record.set_tip(tip);
     }
 
+    fn set_prune_modes(&mut self, prune_modes: PruneModes) {
+        self.batch_record.set_prune_modes(prune_modes);
+    }
```

Comment:
this should not be part of the interface at all.

none of the pruning stuff should happen in the executor itself, this is just weird.